datafusion/common/src/scalar/mod.rs

```diff
@@ -500,6 +508,14 @@ impl PartialOrd for ScalarValue {
             (DurationMicrosecond(_), _) => None,
             (DurationNanosecond(v1), DurationNanosecond(v2)) => v1.partial_cmp(v2),
             (DurationNanosecond(_), _) => None,
+            (Union(v1, t1, _m1), Union(v2, t2, _m2)) => {
```

Comment:
won't this mean that dense unions are compared the same as sparse unions? It seems like we should also compare `m1` and `m2`

Reply:
I wonder if we should have the `UnionMode` in here at all? By default calling `to_array_of_size` will generate a sparse union anyway (and creating a dense union from a scalar value wouldn't make sense to begin with....).

Reply:
My thought was that if the unions have the same logical data, the physical way they choose to store it is irrelevant to their logical comparison.

Reply:
As long as `ScalarValue` comparison semantics follows the corresponding arrow-rs `UnionArray` comparison semantics, it is good with me

Reply:
Looks like that's here: https://github.com/coralogix/arrow-rs/blob/9f20bd8298a52cdde80b0abf19a7582b8b1bfd41/arrow/src/array/equal/union.rs#L120

I'll update to match

Reply:
so we should panic :)

Reply:
I didn't want to go that far...